x/auth/tx/xauthclient.go

```diff
@@ -0,0 +1,157 @@
+// Package tx 's xauthclient.go file is copy-pasted from
+// https://github.com/cosmos/cosmos-sdk/blob/v0.41.3/x/auth/client/query.go
+// It is duplicated as to not introduce any breaking change in 0.41.4, see PR:
+// https://github.com/cosmos/cosmos-sdk/pull/8732#discussion_r584746947
+package tx
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// QueryTxsByEvents performs a search for transactions for a given set of events
+// via the Tendermint RPC. An event takes the form of:
+// "{eventAttribute}.{attributeKey} = '{attributeValue}'". Each event is
+// concatenated with an 'AND' operand. It returns a slice of Info object
+// containing txs and metadata. An error is returned if the query fails.
+// If an empty string is provided it will order txs by asc
+func queryTxsByEvents(clientCtx client.Context, events []string, page, limit int, orderBy string) (*sdk.SearchTxsResult, error) {
+	if len(events) == 0 {
+		return nil, errors.New("must declare at least one event to search")
+	}
+
+	if page <= 0 {
+		return nil, errors.New("page must greater than 0")
+	}
+
+	if limit <= 0 {
+		return nil, errors.New("limit must greater than 0")
+	}
+
+	// XXX: implement ANY
+	query := strings.Join(events, " AND ")
+
+	node, err := clientCtx.GetNode()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: this may not always need to be proven
+	// https://github.com/cosmos/cosmos-sdk/issues/6807
+	resTxs, err := node.TxSearch(context.Background(), query, true, &page, &limit, orderBy)
```

Comment:
let's don't use `context.Background` in this queries. We can get context from `cmd` and provide it through a function argument.